### ISSUE TITLE
Source Code

### ISSUE THREAD

**Issue by robertkirkman** [dshawshank/blender#1]
https://download.blender.org/release/GPL-license.txt

GNU General Public License

Please Send Source Code

**Comment by astroskyoffical** [comment]
I agree source code would allow people to trust this alot more. I'm getting complaints after sharing this to Reddit that they don't trust others without source code. I understand this your personal project but if you share the source code with rules you can likely have others improve it's use case on many devices and add other features.  Helping your idea come to life.   Please consider making this open source like Godot is. This is huge for Android.

**Comment by dshawshank** [comment]
This is a very serious topic, but I am not ready yet, including source code, comments, documentation, terms, support...

**Comment by Cloudwalk9** [comment]
> This is a very serious topic, but I am not ready yet, including source code, comments, documentation, terms, support...

It doesn't matter. You're violating the GPL.

**Comment by Cloudwalk9** [comment]
@astroskyoffical It's not just trust, it's a legal issue because anyone who ever contributed to Blender can DMCA the repo. He can't share the source code with any more rules or restrictions than the GPL itself.

**Comment by astroskyoffical** [comment]
Yeah he might as well share and be in good standing in case that happens

**Comment by Cloudwalk9** [comment]
@astroskyoffical It's not like getting a copyright strike on YouTube. I give the author the benefit of the doubt here but GPL violations should never be tolerated as a rule.

**Comment by gyoo18** [comment]
> This is a very serious topic, but I am not ready yet, including source code, comments, documentation, terms, support...

Sharing source code is not like publishing a release version. Of course nothing will be production ready, or user-friendly, everyone understamds that. Additionally, this is still in alpha and you are working on the project alone (or with a goup, but not with the community), so nothing needs to be production-ready for others to understand and contribute to. It simply needs to be shown.

Of course remove anything compromsing, but don't feel like you need to add comments, fix bugs or refactor code to make it public, you'll do that after. That’s the point of github.

**Comment by Cloudwalk9** [comment]
@gyoo18 He must post the complete, full source code of the APK. He can't omit anything.

**Comment by Cloudwalk9** [comment]
Just concerned about people taking the GPL for granted and trivializing copyright infringement, violating it. Sometimes GPL's reciprocity is the reason we get nice things. No pressure on the author intended but it's still a legal obligation.

**Comment by gyoo18** [comment]
@Cloudwalk9 Of course! Let's simply give the benifit of the doubt and give him the time to show good faith behavior before throwing DMCA takedowns 😜.

**Comment by dshawshank** [comment]
I am a rigorous developer who respects code. 

It is impossible for me to throw out some fragmented codes before clarifying the outline of such a great software as Blender. 

This is a blasphemy to Blender.

**Comment by robertkirkman** [comment]
Now that the maintainer has explained his response, I can now mention this preexisting build of Blender for Android which is Open Source https://github.com/termux-user-repository/tur/tree/master/tur-on-device/blender

https://f-droid.org/en/packages/com.termux/
https://github.com/termux/termux-x11/releases

```bash
pkg upgrade
pkg install x11-repo tur-repo
pkg install termux-x11-nightly xfce blender
export TERMUX_X11_XSTARTUP=xfce4-session DISPLAY=:0
termux-x11 &
blender &
```
Graphics Drivers commonly known to yield usable performance with this:
- `zink` OpenGL Desktop 4.3 driver  (built for X11 wsi bionic libc) layered on top of "Turnip" aka  "`freedreno`"  Vulkan 1.3 driver (built for X11 wsi bionic libc) that comes from `pkg install mesa-vulkan-icd-freedreno`
  - GPU compatibility: **only Adreno 6XX and 7XX GPUs and very recent Mali which can use Zink effectively**. Adreno 5XX and older, and older Mali devices cannot run Blender with GPU connection.

Until such time as the codes of this Blender for Android `ANativeWindow` AKA `/system/bin/surfaceflinger` are less "fragmented and blasphemous to the greatness of such a software as Blender",

This alternative preexisting build of Blender for Android X11 can continue to fulfill the need for fully open source Blender on Android, for the time being.

**Comment by Cloudwalk9** [comment]
@gyoo18 The author calling a GPL violation "a blasphemy to Blender" kinda diminishes good faith imo. Hope this gets DMCA'd. That's the lamest excuse I've ever heard to not respect the GPL and the work hundreds of people volunteered to make this especially in the earlier days.

**Comment by gyoo18** [comment]
I thought they were talking about their code. Anyways, I don't want to continue this debate. I'll personaly wait for at least a week before supporting a dmca takedown.

**Comment by Patataonk** [comment]
thanks for your work on this dshawshank

I hope your work will demonstrate that there should be an official blender android port because tablets these days are more than capable to handle blender

krita has android native, godot has android native, blender android is like the next dream

**Comment by Dynacato** [comment]
@robertkirkman can you make a full tutorial explaining on full detail all the process to get the already existing native port of blender for Android?

**Comment by gyoo18** [comment]
@dshawshank It has been a week and there is still no source code. Please release the source code or we will have to take action.
